### PR TITLE
Page/acq 645  accessibility and code.org

### PR DESCRIPTION
**Pull request by dmantonyuk**
### Update Accessibility page
Old prod link: 
https://code.org/accessibility

## Links
Jira ticket: 
https://codedotorg.atlassian.net/browse/ACQ-645

## Follow-up work

- Created new layout based on figma, text taken from figma and added as translation ready.
- Loops have been added for cards to make the code shorter and clearer

## Privacy
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
        - Not
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
        - Not

## PR Checklist:

- [ ] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dmantonyuk** [comment]
image

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml [hunk 1]
```diff
@@ -297,6 +298,8 @@
   module_label_educator_tool: "Educator tool"
   module_label_educator_course: "Educator course"
   module_label_student_tool: "Student tool"
+  module_label_student_product: "Product"
+  module_label_student_content_resources: "content & resources"
```
Comment (kelbyhawn) [inline pegasus/sites.v3/hourofcode.com/i18n/en.yml]: Nit: can you update this to sentence case (`Content & resources`) so it's like the others?

File: pegasus/sites.v3/code.org/public/accessibility.haml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      .col-50.flex-container.justify-end
+        %img{src: "/images/accessibility/hero-image.png", alt: ""}
+    .clear
+
+%section.accessibility-efforts.bg-neutral-light
+  .wrapper
+    %h2
+      = hoc_s(:accessibility_efforts_title)
+    %p
+      = hoc_s(:accessibility_efforts_desc)
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/accessibility.haml]: It looks like the `p.body-two` class is being applied to this paragraph, but it should be the default 16px/1rem paragraph size.

Screenshot 2023-07-18 at 2 29 11 PM

File: pegasus/sites.v3/code.org/public/accessibility.haml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    .grid-container
+      - item_count = 4
+      - item_count.times do |index|
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/accessibility.haml:31]: Love this! 🙌

File: pegasus/sites.v3/code.org/public/accessibility.haml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                = hoc_s("accessibility_efforts_card_desc_#{index + 1}")
+
+%section.accessibility-explore
+  .wrapper
+    %h2=hoc_s(:accessibility_explore_title)
+    .action-block.action-block--two-col
+      %img{src: "/images/accessibility/explore.jpg", alt: hoc_s(:accessibility_explore_title), class: "col-50"}
+      .text-wrapper.col-50
+        %p.heading-sm=hoc_s(:accessibility_explore_text)
+        %a{href: "https://support.code.org/hc/en-us/sections/15186591538957-Accessibility", class: "link-button", target: "_blank"}
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/accessibility.haml]: For security reasons can you make sure to include `rel: "noopener noreferrer"` on all links that open in a new tab?

File: pegasus/sites.v3/code.org/public/accessibility.haml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+- accessibility_partner_tags = [hoc_s(:module_label_student_content_resources),hoc_s(:module_label_student_product),hoc_s(:module_label_student_product),hoc_s(:module_label_student_content_resources),hoc_s(:module_label_curriculum_pure),hoc_s(:module_label_student_product),hoc_s(:module_label_curriculum_pure),hoc_s(:module_label_student_product),hoc_s(:module_label_student_product),hoc_s(:module_label_curriculum_pure)]
+- accessibility_partner_levels = [": K-12", ": K-8", ": K-6",": K-12", ": K-12", ": K-12", ": K-12", ": K-6", ": K-12", ": K 9-12"]
+
+%section.accessibility-partner
+  .wrapper
+    %h2=hoc_s(:accessibility_partner_title)
+    %p=hoc_s(:accessibility_partner_desc)
+    .action-block__wrapper.action-block__wrapper--three-col
+      - partner_index.times do |index|
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/accessibility.haml:86]: So cool 😎 — I need to try this out on a page soon!

File: pegasus/sites.v3/code.org/public/accessibility.haml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            %p.overline
+              =accessibility_partner_tags[index]
+            %h3=hoc_s("accessibility_card_title_#{index + 1}")
+            %p
+              =hoc_s(:grade_level)
+              =accessibility_partner_levels[index]
+            %p=hoc_s("accessibility_card_text_#{index + 1}")
+          .content-footer
+            %a.link-button{href: accessibility_partner_links[index], target: "_blank"}
+              = hoc_s(:call_to_action_learn_more)
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/accessibility.haml:98]: Can you add aria-labels to the buttons in this section to be more descriptive? Similar to what you did here on the AI PL 101 page: https://github.com/code-dot-org/code-dot-org/blob/8996b0805e8c1b8356736f71cfa33ca0f7db8977/pegasus/sites.v3/code.org/public/ai/pl/101/index.haml#L61

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml [hunk 7]
```diff
@@ -8015,6 +8018,56 @@
...
+  accessibility_hero_heading: "Accessibility at Code.org"
+  accessibility_hero_desc: "We are dedicated to eliminating barriers that prevent the inclusion and full participation of students and educators with disabilities, ensuring that EVERY student has access to computer science."
+  accessibility_efforts_title: "Our Efforts for 2023"
+  accessibility_efforts_desc: "To achieve our goal of a fully accessible platform, we’re implementing a comprehensive action plan that includes:"
+  accessibility_efforts_card_title_1: "Internal accessibility training"
+  accessibility_efforts_card_desc_1: "Our product and engineering teams will complete comprehensive accessibility training by December 2023."
+  accessibility_efforts_card_title_2: "VPAT Analysis"
+  accessibility_efforts_card_desc_2_a: "Code.org will complete an annual"
+  accessibility_efforts_card_desc_2_b: "Voluntary Product Accessibility Template"
+  accessibility_efforts_card_desc_2_c: "to establish a baseline and measure progress annually."
```
Comment (kelbyhawn) [inline pegasus/sites.v3/hourofcode.com/i18n/en.yml]: For strings that have links, be sure to keep it all in one string so things don't get wonky when translated. You can add links inline using this pattern: `[Link copy here](%{link_url})`. 

Here's example of how this works: 
https://github.com/code-dot-org/code-dot-org/blob/8996b0805e8c1b8356736f71cfa33ca0f7db8977/pegasus/sites.v3/hourofcode.com/i18n/en.yml#L707

https://github.com/code-dot-org/code-dot-org/blob/8996b0805e8c1b8356736f71cfa33ca0f7db8977/pegasus/sites.v3/code.org/views/faq_csp.haml#L6

**Review by kelbyhawn — COMMENTED** [review]
Overall this looks great! Thanks for the quick turnaround on this and adding the loop logic. 

Left a couple comments, and I'll take another look once those are updated 😄

File: pegasus/sites.v3/code.org/public/accessibility.haml (hunk 2 above)
Comment (dmantonyuk) [reply pegasus/sites.v3/code.org/public/accessibility.haml]: done

File: pegasus/sites.v3/code.org/public/accessibility.haml (hunk 4 above)
Comment (dmantonyuk) [reply pegasus/sites.v3/code.org/public/accessibility.haml]: done

**Comment by dmantonyuk** [comment]
@kelbyhawn please check again

File: pegasus/sites.v3/code.org/public/accessibility.haml (hunk 6 above)
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/accessibility.haml:98]: Thanks for adding these! Can you update the text though so the button action is more descriptive, like `Learn more about [accessibility_card_title…]`.

File: pegasus/sites.v3/code.org/public/accessibility.haml (hunk 2 above)
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/accessibility.haml]: It's only that first paragraph that needed to be updated — the others in the blocks can remain 14px/0.875rem.

**Comment by kelbyhawn** [comment]
> @kelbyhawn please check again

Thanks for updating! I left two more comments, but once those are done I think it's good to go.

File: pegasus/sites.v3/hourofcode.com/i18n/en.yml (hunk 1 above)
Comment (dmantonyuk) [reply pegasus/sites.v3/hourofcode.com/i18n/en.yml]: done

**Pull request merged**